### PR TITLE
[WIP] Support uda

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Targets #31

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
@mario-kr Do you see my error here? It seems like there is no `deserialize()` for `String`, but the serde docs say `Deserialize` is implemented on `String` ...

File: src/uda.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    }
+
+}
+
+impl Deserialize for UDAName {
+
+    fn deserialize<D>(deserializer: &mut D) -> RResult<UDAName, D::Error>
+        where D: Deserializer
+    {
+        String::deseralize(deserializer).map(|s| UDAName(s))
```
Comment (mario-kr) [inline src/uda.rs]: it's a typo. You should read the error output a bit more cautiously

Comment (mario-kr) [inline src/uda.rs]: you probably facepalm right now xD

Comment (matthiasbeyer) [inline src/uda.rs]: I hate myself right now. :angry: :nerd_face:

**Comment by matthiasbeyer** [comment]
Okay, now it should compile, but all tests fail...

I will have a look at this tomorrow or something, but I guess not today.

**Comment by maralorn** [comment]
Hey, how is this PR going? I previouly had the hazzle to work with udas and task_hookrs, which basically meant, that I forked it and implemented my udas manually.

I feel, like this PR is a bit stale right now. Would you mind, if I picked it up to see if I can get it to work?

**Comment by matthiasbeyer** [comment]
@maralorn Hey, that would be really awesome! Feel free to do so!

It is stalled, yes. Unfortunately the featureset task-hookrs has right now suffices for my own needs and so I focused on other things. I'd love to see contribution on this repository!

**Pull request closed without merging**